airbyte-cdk/python/connector_builder/main.py

```diff
@@ -0,0 +1,67 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+
+import argparse
+import sys
+from typing import Any, List, Mapping, Tuple
+
+from airbyte_cdk.connector import BaseConnector
+from airbyte_cdk.entrypoint import AirbyteEntrypoint, launch
+from airbyte_cdk.sources.declarative.manifest_declarative_source import ManifestDeclarativeSource
+from connector_builder.connector_builder_handler import resolve_manifest
+
+
+def create_source(config: Mapping[str, Any]) -> ManifestDeclarativeSource:
+    manifest = config.get("__injected_declarative_manifest")
+    return ManifestDeclarativeSource(manifest)
+
+
+def get_config_from_args(args: List[str]) -> Mapping[str, Any]:
+    command, config_filepath = preparse(args)
+    if command == "spec":
+        raise ValueError("spec command is not supported for injected declarative manifest")
+
+    config = BaseConnector.read_config(config_filepath)
+
+    if "__injected_declarative_manifest" not in config:
+        raise ValueError(
+            f"Invalid config: `__injected_declarative_manifest` should be provided at the root of the config but config only has keys {list(config.keys())}"
+        )
+
+    return config
+
+
+def preparse(args: List[str]) -> Tuple[str, str]:
```

Comment:
the interface should be the same as other connectors so this should also accept `--catalog` and and optional `--state`

It might make sense to always use `AirbyteEntrypoint.parse_args`

Reply:
@clnoll is there a reason to use a preparse method instead of directly calling `AirbyteEntrypoint.parse_args`? Using the same method ensures we're using the same interface

Reply:
Good call, updated to use `AirbyteEntrypoint.parse_args`.